index.js

```diff
@@ -338,22 +338,30 @@ exports.stringify = (object, options) => {
 };
 
 exports.parseUrl = (input, options) => {
-	return {
-		url: removeHash(input).split('?')[0] || '',
-		query: parse(extract(input), options)
-	};
+	const [url, hash] = splitOnFirst(input, '#');
+	return Object.assign(
+		{
+			url: url.split('?')[0] || '',
+			query: parse(extract(input), options)
+		},
+		options && options.parseFragmentIdentifier && hash ? {fragmentIdentifier: hash} : {}
+	);
 };
 
 exports.stringifyUrl = (input, options) => {
 	const url = removeHash(input.url).split('?')[0] || '';
 	const queryFromUrl = exports.extract(input.url);
 	const parsedQueryFromUrl = exports.parse(queryFromUrl);
-	const hash = getHash(input.url);
+	let hash = getHash(input.url);
 	const query = Object.assign(parsedQueryFromUrl, input.query);
 	let queryString = exports.stringify(query, options);
 	if (queryString) {
 		queryString = `?${queryString}`;
 	}
 
+	if (options && options.parseFragmentIdentifier && input.fragmentIdentifier) {
+		hash = `#${input.fragmentIdentifier}`;
```

Comment:
Shouldn't this be properly encoded? Also needs tests for this to ensure it properly encodes.